transitions/center_wipe.h

```diff
@@ -0,0 +1,88 @@
+#ifndef TRANSITIONS_CENTER_WIPE_H
+#define TRANSITIONS_CENTER_WIPE_H
+
+#include "wave.h"
+#include "../functions/mult.h"
+#include "../functions/sum.h"
+
+// Usage: TrCenterWipeX<POSITION_FUNCTION, MILLIS_FUNCTION>
+// or: TrCenterWipeX<POSITION, MILLIS>
+// POSITION_FUNCTION & MILLIS_FUNCTION: FUNCTION
+// POSITION: Int
+// MILLIS: a number
+// return value: TRANSITION
+// In the beginning entire blade is color A, then color B 
+// starts at the POSTION and extends up and down the blade
+// in the specified number of milliseconds.
+template<class MILLIS, class POSITION = Int<16384>>
+class TrCenterWipeX : public TransitionBaseX<MILLIS> {
+public:
+  void run(BladeBase* blade) {
+    pos_.run(blade);
+    TransitionBaseX<MILLIS>::run(blade);
+    center_ = (pos_.getInteger(0) * blade->num_leds()) >> 7;
+    fade_top_ = this->update(256 * blade->num_leds() - center_);
+    fade_bottom_ = this->update(center_);
+    top_ = clampi32(center_ + fade_top_, center_, 256 * blade->num_leds());
+    bottom_ = clampi32(center_ - fade_bottom_, 0, center_);
+  }
+
+  template<class A, class B>
+  auto getColor(const A& a, const B& b, int led) -> decltype(MixColors(a,b,1,1)) {
+        int mix = (Range(bottom_, top_) & Range(led << 8, (led << 8) + 256)).size();
+    return MixColors(a, b, mix, 8);
+  }
+private:
+  POSITION pos_;
+  uint32_t fade_top_;
+  uint32_t fade_bottom_;
+  uint32_t center_;
+  uint32_t top_;
```

Comment:
replace top_ and bottom_ with: Range range_;

Reply:
I could not get it to compile when trying to replace, how should it be done?

Reply:
Replace top_ and bottom_ with:

   Range range_;

Make top and bottom local variables, and then at the end of Run(), do:

   range_ = Range(bottom, top);

in getColor, replace Range(botom, top) with range_.

same below

Reply:
Range range_; added and tested, thank you.